### PR TITLE
USWDS-Site: Update stylelint to match USWDS

### PR DESCRIPTION
**Pull request by mahoneycm**
## Description

Related to #1512 

## Additional information

Getting build errors related to the change to `sass.js` 

I also tried using this bit of refactored code from Dan's changes

```
gulp.task('stylelint', async function (callback) {
  const { errored, output } = await stylelint.lint({
    files: [`${PROJECT_SASS_SRC}/**/*.scss`, `!${PROJECT_SASS_SRC}/uswds/**/*.scss`],
    formatter: 'string',
  });

  callback(errored ? new Error(output) : null);
});
```
https://github.com/uswds/uswds/pull/4425/files#r768329890

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [ ] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [ ] Run `npm test` and make sure the tests for the files you have changed have passed.
- [ ] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
- [ ] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

### CODE REVIEW

File: config/gulp/sass.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
-const { formatters } = require("stylelint");
+// const { formatters } = require("stylelint");
```
Comment (mejiaj) [inline config/gulp/sass.js:1]: Please remove commented out code.

File: css/_uswds-theme-custom-styles.scss [hunk 2]
```diff
@@ -802,7 +806,7 @@ body {
   }
 }
 
-.preview-no_border {
+.preview-no__border {
```
Comment (mejiaj) [inline css/_uswds-theme-custom-styles.scss:809]: Was this class even used? Not seeing any templates using either.

If it's not being used, let's remove the styles.

File: css/_uswds-theme-custom-styles.scss [hunk 3]
```diff
@@ -880,11 +884,11 @@ body {
   }
 }
 
-h3 + .button_wrapper {
+h3 + .button__wrapper {
```
Comment (mejiaj) [inline css/_uswds-theme-custom-styles.scss:887]: Not finding templates with this class either.

File: css/_uswds-theme-custom-styles.scss [hunk 4]
```diff
@@ -899,7 +903,7 @@ h3 + .button_wrapper {
   }
 }
 
-.button_wrapper-dark {
+.button__wrapper--dark {
```
Comment (mejiaj) [inline css/_uswds-theme-custom-styles.scss:906]: Same with this class, no references found in markup.

File: css/_uswds-theme-custom-styles.scss [hunk 5]
```diff
@@ -1478,7 +1482,7 @@ code[class*="language-plaintext"] {
 
 .usa-button,
 .link-download,
-.media_link {
+.media__link {
```
Comment (mejiaj) [inline css/_uswds-theme-custom-styles.scss:1485]: The class changed, but the markup still references the old. Please update in markup.

File: css/_uswds-theme-custom-styles.scss [hunk 6]
```diff
@@ -1668,7 +1672,7 @@ code[class*="language-plaintext"] {
   margin-left: units(3);
 }
 
-.link_group-download {
+.link__group-download {
```
Comment (mejiaj) [inline css/_uswds-theme-custom-styles.scss:1675]: Class was updated in CSS, but not in markup. I still see templates that use `link_group-download`.

File: css/_uswds-theme-custom-styles.scss [hunk 7]
```diff
@@ -2844,8 +2853,9 @@ $theme-table-column-gap: 4;
       @include u-text("uppercase", "ls-2", "normal");
     }
 
-    .site-text-intro {
-      @extend .site-text-intro;
+    .site-text-intro,
+    %site-text-intro {
+      @extend %site-text-intro;
```
Comment (mejiaj) [inline css/_uswds-theme-custom-styles.scss:2858]: This placeholder selector should be defined before `.site-prose` and then extended here and in line `2893`.

File: css/_uswds-theme-custom-styles.scss [hunk 8]
```diff
@@ -2106,7 +2111,8 @@ li {
   @include u-text("light", "no-uppercase", "ls-auto", "red-warm-50v");
 }
 
-.utility-scope {
+.utility-scope,
+%utility-scope {
```
Comment (mejiaj) [inline css/_uswds-theme-custom-styles.scss:2115]: This extend is created, but never actually used. Do we need this?

File: css/uswds-next.scss [hunk 9]
```diff
@@ -701,15 +701,15 @@ html {
     }
   }
 
-  .next-opportunity__need__need-img {
+  .next-opportunity__need--need-img {
```
Comment (mejiaj) [inline css/uswds-next.scss:704]: If we updated the class, we should also update where it's used in markup.

File: css/uswds-next.scss [hunk 10]
```diff
@@ -701,15 +701,15 @@ html {
...
     @include u-square(2);
     margin-right: units(2);
     @include at-media("desktop") {
       @include u-square(2);
     }
   }
 
-  .next-opportunity__need__need-label {
+  .next-opportunity__need--need-label {
```
Comment (mejiaj) [inline css/uswds-next.scss:712]: Let's make this update in markup too.

**Review by mejiaj — CHANGES_REQUESTED** [review]
Linter task looks good! 

Just added some comments to the newly updated styles.

File: css/_uswds-theme-custom-styles.scss (hunk 2 above)
Comment (mahoneycm) [reply css/_uswds-theme-custom-styles.scss]: Removed in [68a4160](https://github.com/uswds/uswds-site/pull/1573/commits/68a4160c2289b20ae11c9a89a23e97cb59f3b6a4)

**Comment by mahoneycm** [comment]
It looks like there are a lot of unused classes in the `_uswds-theme-custom-styles.scss` file.  Going to make a new issue to address @mejiaj  notes above

**Review by thisisdano — APPROVED** [review]

**Pull request merged**